### ISSUE TITLE
OutOfMemoryException: Out of memory

### ISSUE THREAD

**Issue by bloodythorn** [Theo47/depressurizer#71]
v0.7.1.3 crashes intermittently while using it. It's something that seems to be happening randomly, as the only way to re-create it is to use it for any length of time until it crashes. It seems to have crashed every attempted use so far.

I was reading through closed issues, and it seems similar, if not the same as closed issue #61.

[dError_20161029043103.log.txt](https://github.com/Theo47/depressurizer/files/560190/dError_20161029043103.log.txt)

Tell me if you need anything else.

**Comment by kazerniel** [comment]
I've just downloaded v0.7.1.4 and it crashed with this error:

```
OutOfMemoryException: Out of memory.
   at System.Drawing.Image.FromFile(String filename, Boolean useEmbeddedColorManagement)
   at Depressurizer.FormMain.lstGames_FormatCell(Object sender, FormatCellEventArgs e)
   at BrightIdeasSoftware.ObjectListView.OnFormatCell(FormatCellEventArgs args)
   at BrightIdeasSoftware.ObjectListView.TriggerFormatCellEvents(Int32 rowIndex, Int32 displayIndex, OLVListItem olvi)
   at BrightIdeasSoftware.ObjectListView.PostProcessOneRow(Int32 rowIndex, Int32 displayIndex, OLVListItem olvi)
   at BrightIdeasSoftware.VirtualObjectListView.MakeListViewItem(Int32 itemIndex)
   at BrightIdeasSoftware.VirtualObjectListView.HandleRetrieveVirtualItem(Object sender, RetrieveVirtualItemEventArgs e)
   at System.Windows.Forms.ListView.OnRetrieveVirtualItem(RetrieveVirtualItemEventArgs e)
   at System.Windows.Forms.ListView.ListViewNativeItemCollection.get_Item(Int32 displayIndex)
   at System.Windows.Forms.ListView.ListViewItemCollection.get_Item(Int32 index)
   at System.Windows.Forms.ListView.WmReflectNotify(Message& m)
   at System.Windows.Forms.ListView.WndProc(Message& m)
   at BrightIdeasSoftware.ObjectListView.WndProc(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.OnMessage(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)
```

(I have 707 games if that's relevant, and was trying to AutoCat 3 of them :) )

**Comment by hymerman** [comment]
Same here. Happens for me with 700+ games, selecting all and removing a tag straight after startup, or selecting all and using AutoCat sometimes after a while of running.

**Comment by StarWeaver** [comment]
I was getting this a week or so ago too. After categorizing some things out of <uncategorized>, it started happening every time I launched; it seems to have something to do with a corrupt or nonexistent image file for one of my games that got closer and closer to the top of the list? Not really sure.

This was with 0.7.1.4. I tried 0.7.1.0 instead and it worked fine, but couldn't see a dozen or two of my games.

**Comment by hymerman** [comment]
Got it. Looks like the `Image.Save` call used to save downloaded images is sometimes corrupting them - downloading the same URL from Chrome, or saving the image using another method, is fine.

I've tweaked the code to use another method to save the files, which should result in a few of the problematic ones now being supported where they weren't before (e.g. AppId 247000). I've also added a try/catch block around the image loading code to stop it crashing in cases where broken images are already cached.

I'll fork and submit a PR shortly.
